src/OpenTelemetry.Api/Logs/LoggerProvider.cs

```diff
@@ -0,0 +1,53 @@
+// <copyright file="LoggerProvider.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#nullable enable
+
+namespace OpenTelemetry.Logs;
+
+/// <summary>
+/// LoggerProvider is the entry point of the OpenTelemetry API. It provides access to <see cref="Logger"/>.
+/// </summary>
+public class LoggerProvider : BaseProvider
+{
+    private NoopLogger? noopLogger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LoggerProvider"/> class.
+    /// </summary>
+    protected LoggerProvider()
+    {
+    }
+
+    internal Logger GetLogger()
+        => this.GetLogger(name: null);
+
+    internal Logger GetLogger(string? name)
+    => this.GetLogger(new LoggerOptions(name));
+
+    internal Logger GetLogger(InstrumentationScope instrumentationScope)
```

Comment:
Since metrics and traces do not yet support `InstrumentationScope` would it make sense to just have 

```csharp
internal Logger GetLogger(string name, string version)
```

This mirrors .NET's support for meters and activities. When the runtime introduces support for schema url and attributes, we should follow the same pattern for the logger.

Also, I understand InstrumentationScope is internal right now, so not a big deal if this merges as is.

Reply:
This one was kind of an interesting journey ⛵

Check out: [Get a Logger](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/api.md#get-a-logger). There are a couple unique parameters `event_domain` (required if you want to emit events) & `include_trace_context` (optional, true by default). I think the first natural thing to do is like...

```csharp
internal Logger GetLogger(string? name = null, string? version = null, string? eventDomain = null, bool includeTraceContext = true) {}
```

But then what do we do if we want to support the other scope things (`schema_url`, `attributes`) in the future? We have to drop in another giant overload, or have a whole bunch of non-defaulted versions to try and cover all the permutations. If the spec ever adds anything else, more trouble.

So in order to keep it simple and allow for expansion I originally just had:

```csharp
Logger GetLogger(LoggerOptions options) {}
```

```csharp
internal sealed class LoggerOptions
{
    public LoggerOptions(string? name) {}

    public string Name { get; }
    public string? Version { get; init; }
    public string? SchemaUrl { get; init; }
    public IReadOnlyDictionary<string, object>? Attributes { get; init; }
    public string? EventDomain { get; init; }
    public bool IncludeTraceContext { get; init; } = true;
}
```

But then, `Name` + `Version` + `SchemaUrl` + `Attributes` are really "instrumentation scope" which (by the spec) also apply to tracing & metrics. So to achieve potential re-use it became...

```csharp
internal sealed class LoggerOptions
{
    public LoggerOptions(InstrumentationScope instrumentationScope) {}

    public InstrumentationScope InstrumentationScope { get; }
    public string? EventDomain { get; init; }
    public bool IncludeTraceContext { get; init; } = true;
}
```

I was thinking at some point we might do...

```diff
class LoggerProvider
{
    public Tracer GetTracer(string name, string version = null) {}
+   public Tracer GetTracer(InstrumentationScope instrumentationScope) {}
}
```

Then I started to write tests and update the code to use this API. When you just want to pass a name having to new up `LoggerOptions` I felt was too ceremonious. So I add some helpers for the common cases.

```csharp
    internal Logger GetLogger()
        => this.GetLogger(name: null);

    internal Logger GetLogger(string? name)
    => this.GetLogger(new LoggerOptions(name));

    internal Logger GetLogger(InstrumentationScope instrumentationScope)
        => this.GetLogger(new LoggerOptions(instrumentationScope));
```

Worth noting here is [the `InstrumentationScope` does end up on `LogRecord`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/8174837837a58c8c6c4a9aa7123f21463aca0b80/src/OpenTelemetry/Logs/LogRecord.cs#L263) so exporters _could_ actually be authored to be spec complaint w.r.t. "instrumentation scope".

So what should we do here? 🤷 It is all internal so we could just wait and see 😄

Reply:
> It is all internal so we could just wait and see 😄

Yes. I think it would be ideal if these things were encapsulated in a .NET class not an OpenTelemetry class. Relevant to https://github.com/dotnet/runtime/issues/63651 - the API here should be expanded to include the scope attributes.